internals/slo.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import logging
+import pytz
+
+from framework import permissions
+from framework.users import User
+from internals.core_models import FeatureEntry
+from internals.review_models import Gate, Vote
+
+PACIFIC_TZ = pytz.timezone('US/Pacific')
+MAX_DAYS = 9999
+
+
+def is_weekday(d: datetime.datetime) -> bool:
+  """Return True if d is a weekday: Monday through Friday."""
+  return d.weekday() < 5
+
+
+def weekdays_between(start: datetime.datetime, end: datetime.datetime) -> int:
+  """Return the number of Pacific timezone weekdays between two UTC dates."""
+  d_ptz = start.astimezone(PACIFIC_TZ)
+  # The day of the request does not count.
+  d_ptz = d_ptz.replace(hour=23, minute=59, second=59)
+  end_ptz = end.astimezone(tz=PACIFIC_TZ)
+  weekday_counter = 0
+  while d_ptz < end_ptz and weekday_counter < MAX_DAYS:
+    d_ptz = d_ptz + datetime.timedelta(days=1)
```

Comment:
if d_ptz is a Friday and end_ptz is saturday, I suspect this function will return  0. Should be move `d_ptz = d_ptz + datetime.timedelta(days=1)`  after the `is_weekday(d_ptz)` check?

Reply:
I think returning 0 is correct: the day of the request does not count, and Saturday is a weekend, so neither one of them should increase the count.